### ISSUE TITLE
Use cozy (sud-)domain to send reports

### ISSUE THREAD

**Issue by ZeHiro** [bnjbvr/kresus#284]
Current reports are sent from kresus with kresus-noreply@cozycloud.cc
The email should be emitted by adress kresus-noreply@my.cozy.subdomains.cc as the ip/domain do not match they can be refused.

**Comment by ZeHiro** [comment]
I think I can correct this one.

**Comment by bnjbvr** [comment]
Indeed. However, sending from the host can have some other unintended consequences (if there's a mail server on the machine, for instance, and the "kresus" user doesn't exist, then the mail will be rejected). Ideally, one should be able to provide the email address from which the email is sent, as well as provide credentials for a SMTP access. This would also solve the issue for self-hosted people, who don't have access to the email feature yet.

1. So, a first step would be to detect the hostname and craft the sender email from this: if the cozy is hosted on cozy.zehiro.com, then the email would be kresus@cozy.zehiro.com. This can be done thanks to the CozyInstance data, which ought to contain the domain value.
1. Then, one should be able to change this default email setting in the preferences section.
1. Finally, one should be able to provide SMTP credentials to send from a given email address. The I'm pretty sure there are a few SMTP libraries on npm which would help here. In the worst case, the SMTP protocol isn't too hard to understand.

Feel free to tackle only a subset of these problems. If you do take several ones, please try to do one commit per item :-)

**Comment by ZeHiro** [comment]
For 2 and 3, as I understood cozy was planning to allow to select the smtp server to send the platform  emails (ping @clochix )  we should wait for this feature. Hoping that this would be available to other apps of the platform. 
Then, I would only deal with point 1.

**Comment by clochix** [comment]
Yes, allow to select the SMTP server used by the whole platform is on our long time roadmap (contributions willkommen ;-) ).

I don't understand the initial issue: if you use CozyDB.SendMailFromUser, the sender address should be zehiro-noreply chez zehiro.cozycloud.cc.

**Comment by ZeHiro** [comment]
I think the problem is here, @bnjbvr  didn't use this (but directly connected to the DS).

**Comment by ZeHiro** [comment]
http://cozy.github.io/cozy-db/doc/classes/Api.html#sendMailToUser-instance

**Comment by bnjbvr** [comment]
This won't work in standalone mode, as cozy-db doesn't implement sendMailToUser in this mode. However, this is something we already check in the emails sender, so we can totally use sendMailToUser here. Thanks for the tip!

By the way, I think https://github.com/cozy/cozy-db/blob/master/src/api.coffee#L147 should read from, not to, cc @romaincozy

**Comment by aenario** [comment]
correct, thanks for reporting :)

**Comment by ZeHiro** [comment]
Finally some emails arrive (I think some ports were closed on my ISP-box). But still needs to be fixed.

**Comment by bnjbvr** [comment]
@clochix, `sendMailToUser` needs the API user to set the `from` address. In `sendMailFromUser`, the address is automatically set. Ideally, I'd need something to send an email from the user *to* the user here. In the meanwhile, we can't use `sendMailToUser` or `sendMailFromUser` without having to do some work by ourselves to figure out what the domain name / user email is.

**Comment by clochix** [comment]
@bnjbvr very pertinent comment, could you open an issue in cozy/cozy-data-system ?
